Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -472,10 +483,24 @@ - (void)updateBarButtonItems
     }
 
     //  Settings button.
-    //
-    // TODO: Theme
-    UIImage *image = [UIImage imageNamed:@"button_settings_white"];
-    UIBarButtonItem *settingsButton = [[UIBarButtonItem alloc] initWithImage:image style:UIBarButtonItemStylePlain target:self action:@selector(settingsButtonPressed:)];
+    const NSUInteger kAvatarSize = 28;
+    // TODO: Replace this icon.
+    UIImage *_Nullable localProfileAvatarImage = [OWSProfileManager.sharedManager localProfileAvatarImage];
+    UIImage *avatarImage = (localProfileAvatarImage
+            ?: [[UIImage imageNamed:@"profile_avatar_default"]
+                   imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]);
+    OWSAssertDebug(avatarImage);
+
+    AvatarImageView *avatarView = [[AvatarImageView alloc] initWithImage:avatarImage];
+    if (!localProfileAvatarImage) {
+        avatarView.tintColor = Theme.middleGrayColor;
+    }
+    [avatarView autoSetDimension:ALDimensionWidth toSize:kAvatarSize];
+    [avatarView autoSetDimension:ALDimensionHeight toSize:kAvatarSize];
+    UIBarButtonItem *settingsButton = [[UIBarButtonItem alloc] initWithCustomView:avatarView];
+    avatarView.userInteractionEnabled = YES;
+    [avatarView addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                             action:@selector(settingsButtonPressed:)]];
```

Comment:
We need to verify that this doesn't make it any harder to hit the app settings button.

One alternative would be to pre-render the avatar view to a `UIImage`.

Reply:
We also lose the "touch down" effect this way. You could try doing something like `DisappearingTimerConfigurationView` which simulates this state.

Also, AvatarImageView is a pretty tiny class - did you explore re-implementing it's features in a UIButton?

Reply:
> DisappearingTimerConfigurationView uses long press which has multiple states. Tap gesture recognizers don't provide anything except the "tapped" state.

Right. I was suggesting we do more than a simple tap gesture, which is insufficient for providing the UX feedback that UIButtons offer.